### PR TITLE
Features compilation fix

### PR DESCRIPTION
**Pull request by manu0466**
This PR fixes some compilation errors when compiling `desmos-std` without the default features and adds some CI tests to prevent regressions

### CODE REVIEW

File: packages/desmos-std/src/relationships/mod.rs [hunk 1]
```diff
@@ -1,9 +1,11 @@
-#[cfg(not(target_arch = "wasm32"))]
+#[cfg(all(not(target_arch = "wasm32"), feature = "query"))]
```
Comment (dadamu) [inline packages/desmos-std/src/relationships/mod.rs:1]:
```suggestion
#[cfg(all(not(target_arch = "wasm32"), feature = "mocks"))]
```

File: .github/workflows/desmos_std.yml [hunk 2]
```diff
@@ -75,3 +75,58 @@ jobs:
...
+
+      - name: Build feature (relationships)
+        run: cargo build --no-default-features --features relationships
+        working-directory: ${{env.working-directory}}
+
+      - name: Build feature (relationships+query)
+        run: cargo build --no-default-features --features relationships,query
+        working-directory: ${{env.working-directory}}
+
+      - name: Build feature (msg)
```
Comment (dadamu) [inline .github/workflows/desmos_std.yml:126]: Adding build feature for each modules msg would be useful, say:

```yml
- name: Build feature (profiles+msg)
  run: cargo build --no-default-features --features profiles,msg
  working-directory: ${{env.working-directory}}
```

File: .codecov.yml [hunk 3]
```diff
@@ -39,3 +39,4 @@ ignore:
   - "src/lib.rs"
   - "src/types.rs"
   - "src/state.rs"
+  - "packages/desmos-std/**/mock*.rs"
```
Comment (dadamu) [inline .codecov.yml:42]: Is there any reason to ignore it?

Comment (manu0466) [reply .codecov.yml:42]: Since are mocks I don't think is actually useful to include that code in the codecov report

Comment (bragaz) [reply .codecov.yml:42]: I think it's safe to ignore them since they're only used for testing

**Review by bragaz — COMMENTED** [review]
After fixing @dadamu requests it should be good to be merged

File: .codecov.yml (hunk 3 above)
Comment (dadamu) [reply .codecov.yml:42]: Okay, got it. 👍

**Review by bragaz — APPROVED** [review]
Good to go! 🚀

**Review by dadamu — APPROVED** [review]
🚀

**Pull request merged**